ls1intum/Artemis
# Feature: Improve Presentation Score

### Checklist

### Motivation and Context
This implementation should extend the presentation score feature, described in #865

### Description
- It is now possible to configure presentation scores for courses, i.e. how many presentations a student has to prepare in order to receive the course bonus. This can be done from the `create` / `edit` course component. The instructor has to tick the checkbox and enter the presentation score. (Screenshot 1)
- This presentation score is - when set - displayed in the `course-detail` component and in the `course-statistics` component (Screenshot 2 & 3)
- When the course has a presentation score, then the instructor can define for every single exercise (except for `QUIZ` exercises), if it should be enabled. If yes, then the tutors can `add a presentation` for students by going to `Course Administration` - `Exercises` - `Participations` (Screenshot 5)
- When the course has a presentation score then the exercises tables will display if this feature is enabled for an exercise or not. (Screenshot 6)

### Steps for Testing

1. Create a new course with presentation score or use the [course 43](https://artemistest.ase.in.tum.de/#/course/43/view) to start testing
2. Create an exercise with and without presentation score enabled or use the existing exercises and enable/disable the feature
3. Check the course-details, exercise-details, participations and course-statistics, if all the values are correctly displayed

### Screenshots
Screenshot 1:
<img width="1440" alt="Screenshot 2019-10-04 at 18 22 42" src="https://user-images.githubusercontent.com/33764166/66223862-d0ad8a00-e6d4-11e9-8d17-205483df6a04.png">
Screenshot 2:
<img width="1440" alt="Screenshot 2019-10-03 at 17 44 51" src="https://user-images.githubusercontent.com/33764166/66143120-f0767c80-e606-11e9-93d2-71fd85832321.png">
Screenshot 3:
<img width="1440" alt="Screenshot 2019-10-03 at 17 46 51" src="https://user-images.githubusercontent.com/33764166/66143154-f8ceb780-e606-11e9-9de3-25ca4e938db4.png">
Screenshot 4:
<img width="1440" alt="Screenshot 2019-10-04 at 18 23 35" src="https://user-images.githubusercontent.com/33764166/66223900-e622b400-e6d4-11e9-9ca3-ac01070c2951.png">
Screenshot 5:
<img width="1440" alt="Screenshot 2019-10-03 at 17 47 11" src="https://user-images.githubusercontent.com/33764166/66143546-9b873600-e607-11e9-8fd2-3f909ae4e2bf.png">
Screenshot 6:
<img width="1440" alt="Screenshot 2019-10-03 at 17 46 25" src="https://user-images.githubusercontent.com/33764166/66143503-8a3e2980-e607-11e9-9c12-7369fe5f1b36.png">
